core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreferWritePartitioning.java

```diff
@@ -0,0 +1,50 @@
+package io.trino.sql.planner.iterative.rule;
+
+import io.trino.Session;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.sql.planner.iterative.Rule;
+import io.trino.sql.planner.plan.TableWriterNode;
+
+import java.util.Optional;
+
+import static io.trino.SystemSessionProperties.isUsePreferredWritePartitioning;
+import static io.trino.sql.planner.plan.Patterns.tableWriterNode;
+
+public class PreferWritePartitioning
```

Comment:
Please add a basic rule test, see `io.trino.sql.planner.iterative.rule.TestApplyTableScanRedirection` how we mock table handles in rule tests.